bigquery/google/cloud/bigquery/external_config.py

```diff
@@ -412,46 +542,98 @@ def options(self):
+        """
+        return self._properties.get('compression')
 
-    schema = _ListApiResourceProperty('schema', 'schema', SchemaField)
-    """See
-    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).schema
-    https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.schema
-    """
+    @compression.setter
+    def compression(self, value):
+        self._properties['compression'] = value
+
+    @property
+    def ignore_unknown_values(self):
+        """bool: If `True`, extra values that are not represented in the table
+        schema are ignored. Defaults to `False`.
+
+        See
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).ignoreUnknownValues
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.ignoreUnknownValues
+        """
+        return self._properties.get('ignoreUnknownValues')
+
+    @ignore_unknown_values.setter
+    def ignore_unknown_values(self, value):
+        self._properties['ignoreUnknownValues'] = value
+
+    @property
+    def max_bad_records(self):
+        """int: The maximum number of bad records that BigQuery can ignore when
+        reading data.
+
+        See
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).maxBadRecords
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.maxBadRecords
+        """
+        return self._properties.get('maxBadRecords')
+
+    @max_bad_records.setter
+    def max_bad_records(self, value):
+        self._properties['maxBadRecords'] = value
+
+    @property
+    def source_uris(self):
+        """List[str]: URIs that point to your data in Google Cloud.
+
+        See
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).sourceUris
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.sourceUris
+        """
+        return self._properties.get('sourceUris', [])
+
+    @source_uris.setter
+    def source_uris(self, value):
+        self._properties['sourceUris'] = value
+
+    @property
+    def schema(self):
+        """List[google.cloud.bigquery.schema.SchemaField]: The schema for the
+        data.
+
+        See
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).schema
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.schema
+        """
+        prop = self._properties.get('schema', {})
+        return [SchemaField.from_api_repr(field)
+                for field in prop.get('fields', [])]
+
+    @schema.setter
+    def schema(self, value):
+        prop = value
+        if value:
```

Comment:
I think we probably want this to explicitly check for `is None`. If an empty iterable (`()` or `[]`) is passed in we don't want it being set as the schema (then we'd miss the `{'fields': ...}` part.

Also, Circle says we should add a unit test for passing in None. (We probably want a unit test for passing in `[]`, too).